packages/maskbook/src/social-network-adaptor/twitter.com/injection/Avatar.tsx

```diff
@@ -0,0 +1,115 @@
+import { DOMProxy, LiveSelector, MutationObserverWatcher } from '@dimensiondev/holoflows-kit'
+import { NFTAvatarAmountIcon } from '@masknet/icons'
+import { makeStyles } from '@masknet/theme'
+import { resolveOpenSeaLink } from '@masknet/web3-shared'
+import { Link, Typography } from '@material-ui/core'
+import { AvatarMetaDB, getNFTAvatar } from '../../../components/InjectedComponents/NFT/NFTAvatar'
+import { createReactRootShadowed, Flags, startWatch } from '../../../utils'
+import { postAvatarsContentSelector } from '../utils/selector'
+import { getAvatarId } from '../utils/user'
+
+const useStyles = makeStyles()({
+    root: {
+        display: 'flex',
+        justifyContent: 'center',
+        position: 'absolute',
+        left: 0,
+        top: 44,
+    },
+    wrapper: {
+        position: 'absolute',
+        left: 0,
+        top: 10,
+        background:
+            'linear-gradient(106.15deg, #FF0000 5.97%, #FF8A00 21.54%, #FFC700 42.35%, #52FF00 56.58%, #00FFFF 73.01%, #0038FF 87.8%, #AD00FF 101.49%, #FF0000 110.25%)',
+        borderRadius: 3,
+        minWidth: 43,
+        width: 'auto',
+        display: 'flex',
+        justifyContent: 'center',
+    },
+    icon: {
+        width: '100%',
+        paddingLeft: 10,
+    },
+    text: {
+        fontSize: 10,
+        transform: 'scale(0.8)',
+        margin: 0,
+        color: 'white',
+        whiteSpace: 'nowrap',
+        textShadow:
+            '1px 1px black, 1px 0px black, 0px 1px black, -1px 0px black, 0px -1px black, -1px -1px black, 1px -1px black, -1px 1px black',
+        lineHeight: 1,
+    },
+})
+
+interface NFTBadgeInTwitterProps {
+    avatar: AvatarMetaDB
+}
+function NFTBadgeInTwitter({ avatar }: NFTBadgeInTwitterProps) {
+    const { classes } = useStyles()
+    return (
+        <div
+            className={classes.root}
+            onClick={(e) => {
+                e.preventDefault()
+                window.open(resolveOpenSeaLink(avatar.address, avatar.tokenId), '_blank')
```

Comment:
Remove open in JavaScript. Please use `<Link />` instead.

Reply:
Link cannot jump